cpp/build-support/trim-boost.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+set -eu
+
+BOOST_VERSION=1.71.0
+# These are the ones Arrow uses directly
+BOOST_LIBS="regex.hpp"
+# Arrow tests require these
+BOOST_LIBS="system.hpp filesystem.hpp"
+# Add these to be able to build those
+BOOST_LIBS="$BOOST_LIBS config build boost_install headers"
+# Maybe log is only needed for debug build? (predef is needed for log)
+BOOST_LIBS="$BOOST_LIBS log predef"
```

Comment:
Where do you see this?

Reply:
If I build (a debug build at least, haven't tried otherwise yet), boost_ep build step fails and the error is something like:

```
/Users/enpiar/Documents/ursa/arrow2/cpp/build/boost_ep-prefix/src/boost_ep/tools/build/src/build/project.jam:259: in find-jamfile from module project
error: Unable to load Jamfile.
error: Could not find a Jamfile in directory '/Users/enpiar/Documents/ursa/arrow2/cpp/build/boost_ep-prefix/src/boost_ep/libs/log/config/xopen-source-600'.
error: Attempted to find it with pattern '[Bb]uild.jam [Jj]amfile.v2 [Jj]amfile [Jj]amfile. [Jj]amfile.jam'.
error: Please consult the documentation at 'http://www.boost.org'.
```

Adding `log` to the bundle results in an error about `predef`, and google says that `log` required `predef`.